index.js

```diff
@@ -11,17 +11,16 @@ var resolve = require('resolve')
 var getPackageInfo = require('get-package-info')
 var common = require('./common')
 
-var supportedArchs = {
-  ia32: 1,
-  x64: 1
-}
-
-var supportedPlatforms = {
-  // Maps to module ID for each platform (lazy-required if used)
-  darwin: './mac',
-  linux: './linux',
-  win32: './win32'
-}
+var supportedArchs = common.archs.reduce(function (o, v) {
+  o[v] = 1
+  return o
+}, {})
+
+// Maps to module ID for each platform (lazy-required if used)
+var supportedPlatforms = common.platforms.reduce(function (o, v) {
+  o[v] = './' + (v === 'darwin' ? 'mac' : v)
+  return o
+}, {})
```

Comment:
I think with your current code, the MAS change will look like this:

```javascript
result[platform] = './' + (['darwin', 'mas'].indexOf(platform) !== -1 ? 'mac' : platform)
```

It's not great.